Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -61,61 +61,97 @@ Scott topology, preorder
 -/
 def DirSupClosed (s : Set α) : Prop :=
   ∀ ⦃d⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a⦄, IsLUB d a → d ⊆ s → a ∈ s
 
-@[simp] lemma dirSupInacc_compl : DirSupInacc sᶜ ↔ DirSupClosed s := by
-  simp [DirSupInacc, DirSupClosed, ← not_disjoint_iff_nonempty_inter, not_imp_not,
+@[simp] lemma dirSupClosedOn_univ : DirSupClosedOn univ s ↔ DirSupClosed s := by
+  simp [DirSupClosedOn, DirSupClosed]
+
+@[simp] lemma DirSupClosed.dirSupClosedOn :
+    DirSupClosed s → DirSupClosedOn D s := fun h _ _ d₂ d₃ _ hda => h d₂ d₃ hda
+
+@[simp] lemma dirSupInaccOn_compl : DirSupInaccOn D sᶜ ↔ DirSupClosedOn D s := by
+  simp [DirSupInaccOn, DirSupClosedOn, ← not_disjoint_iff_nonempty_inter, not_imp_not,
     disjoint_compl_right_iff]
 
-@[simp] lemma dirSupClosed_compl : DirSupClosed sᶜ ↔ DirSupInacc s := by
-  rw [← dirSupInacc_compl, compl_compl]
+@[simp] lemma dirSupClosedOn_compl : DirSupClosedOn D sᶜ ↔ DirSupInaccOn D s := by
+  rw [← dirSupInaccOn_compl, compl_compl]
 
-alias ⟨DirSupInacc.of_compl, DirSupClosed.compl⟩ := dirSupInacc_compl
-alias ⟨DirSupClosed.of_compl, DirSupInacc.compl⟩ := dirSupClosed_compl
+alias ⟨DirSupInaccOn.of_compl, DirSupClosedOn.compl⟩ := dirSupInaccOn_compl
+alias ⟨DirSupClosedOn.of_compl, DirSupInaccOn.compl⟩ := dirSupClosedOn_compl
+
+alias ⟨DirSupInacc.of_compl, DirSupClosed.compl⟩ := dirSupInaccOn_compl
+alias ⟨DirSupClosed.of_compl, DirSupInacc.compl⟩ := dirSupClosedOn_compl
 
 lemma DirSupClosed.inter (hs : DirSupClosed s) (ht : DirSupClosed t) : DirSupClosed (s ∩ t) :=
   fun _d hd hd' _a ha hds ↦ ⟨hs hd hd' ha <| hds.trans inter_subset_left,
     ht hd hd' ha <| hds.trans inter_subset_right⟩
 
-lemma DirSupInacc.union (hs : DirSupInacc s) (ht : DirSupInacc t) : DirSupInacc (s ∪ t) := by
-  rw [← dirSupClosed_compl, compl_union]; exact hs.compl.inter ht.compl
+lemma DirSupClosedOn.inter (hs : DirSupClosedOn D s) (ht : DirSupClosedOn D t) :
+    DirSupClosedOn D (s ∩ t) :=
+  fun _d hd₁ hd hd' _a ha hds ↦ ⟨hs hd₁ hd hd' ha <| hds.trans <| inter_subset_left,
+    ht hd₁ hd hd' ha <| hds.trans <| inter_subset_right⟩
+
+lemma DirSupInaccOn.union (hs : DirSupInaccOn D s) (ht : DirSupInaccOn D t) :
+    DirSupInaccOn D (s ∪ t) := by
+  rw [← dirSupClosedOn_compl, compl_union]; exact hs.compl.inter ht.compl
 
-lemma IsUpperSet.dirSupClosed (hs : IsUpperSet s) : DirSupClosed s :=
-  fun _d ⟨_b, hb⟩ _ _a ha hds ↦ hs (ha.1 hb) <| hds hb
+lemma IsUpperSet.dirSupClosedOn (hs : IsUpperSet s) : DirSupClosedOn D s :=
+  fun _d _ ⟨_b, hb⟩ _ _a ha hds ↦ hs (ha.1 hb) <| hds hb
 
-lemma IsLowerSet.dirSupInacc (hs : IsLowerSet s) : DirSupInacc s := hs.compl.dirSupClosed.of_compl
+lemma IsLowerSet.dirSupInaccOn (hs : IsLowerSet s) : DirSupInaccOn D s :=
+  hs.compl.dirSupClosedOn.of_compl
 
-lemma dirSupClosed_Iic (a : α) : DirSupClosed (Iic a) := fun _d _ _ _a ha ↦ (isLUB_le_iff ha).2
+lemma dirSupClosedOn_Iic (a : α) : DirSupClosedOn D (Iic a) :=
+  fun _d _ _ _ _a ha ↦ (isLUB_le_iff ha).2
```

Comment:
I would think the previous lemma was better. When we have an injective function, we prove `Injective f`, not `InjOn f s` for all `s`

Reply:
It's not ideal. I think there are a few options:

- We could just close this PR if it's not useful, and live with having two separate Scott Topologies developed in different ways in `Topology/Order/ScottTopology` and `Topology/OmegaCompletePartialOrder`.
- I add a new file `Topology/Order/ωScottTopology` and replicate the contents of `Topology/Order/ScottTopology` into it but using chains instead of directed sets. This would result in more natural looking statements but a lot of _mutatis mutandis_ duplication.
- Possibly you just meant that you want me to recover the original lemma from this new result?
- Perhaps you have another suggestion? 

I don't have an interest in the `OmegaCompletePartialOrder` / Chain case, so I don't really mind.

Reply:
@YaelDillies did you have any further thoughts on this?

Reply:
I don't understand your comment. All I'm saying is that you should keep the original lemma:
```suggestion
lemma dirSupClosed_Iic (a : α) : DirSupClosed (Iic a) := fun _d _ _ _a ha ↦ (isLUB_le_iff ha).2
```

Reply:
I have restored this lemma.